### ISSUE TITLE
Thank you + RNN's hidden state's expansion factor 

### ISSUE THREAD

**Issue by leofeng-ca** [lucidrains/minGRU-pytorch#1]
Thank you for making the work accessible! I'm sorry about the several typos in the Appendix. Although you have already fixed them in the code. Notably, a few typos in the parallel scan and norms between the layers. 

Some slight differences

1. We included an expansion factor hyperparameter, allowing the RNN's hidden state larger than its input dimension if needed `self.to_hidden_and_gate = Linear(dim, dim * 2 * expansion_factor, bias = False)` and then shrinking it back down to `dim` after the parallel scan.
2. Added a convolutional layer before the RNN for language modelling

**Comment by lucidrains** [comment]
@leofeng-ca hey Leo, thanks for the investigatory work

i added the expansion factor just now, and will get around to that extra convolution this weekend

**Comment by vasqu** [comment]
@leofeng-ca Fmi, so it's using the Mamba1 core block architecture and only replaces the SSM with the minGRU/minLSTM? Is it also applying additional normalization like Mamba2 for example? Thx in advance and great work (to both of you)!

**Comment by leofeng-ca** [comment]
For the small language model experiments, we kept it simple using a Transformer-style block, adding a convolutional layer before the RNN: `norm -> convolutional layer -> minGRU -> 2-layer MLP` with residual connections. We used layer norm and this repo uses rmsnorm but either should work.

**Comment by lucidrains** [comment]
@leofeng-ca hi Leo, threw in that convolution just now

best with the paper
